app/helpers/user_helper.rb

```diff
@@ -0,0 +1,11 @@
+module UserHelper
+  def link_to_user(user_id)
+    return if user_id == 'registrar'
+    
+    if (user = User.find_by(id: user_id))
+      link_to user.kerberos_id, edit_admin_user_path(user.id)
+    else
+      "User ID #{user_id} no longer active."
```

Comment:
As with many Rails conventions, I'm not sure there's an ideal dividing line. I think the general idea is to avoid writing complex logic in views, but what constitutes complex logic is up to interpretation. The other consideration is reusability, which is a factor now that we are using this logic in two places (the hold history view and the reports).

If we moved the markup generation to the view and limited the scope of this helper to returning metadata, we could refactor it to something like this:
```
def return_kerb(user_id)
    return if user_id == 'registrar'

    user = User.find_by(id: user_id)
    return "User ID #{user_id} no longer active." if user.nil?

    user.kerberos_id
end
```

The concern I have with that approach is we would need to introduce conditional logic into the view to determine whether the method call returns an actual kerb, a "User x is no longer active", or nil. An alternative approach could be to have this method _just_ return the kerb and have a separate helper method called `is_active_user?`, but I think we'd still end up delegating the work to the view.

I think I'm running out of steam, but I'll take another look at this tomorrow morning to see if there's a clearer path that I'm missing.

Reply:
Yeah, none of this feels like it is clearly superior to what you've already built. Looking again at the syntax of how this gets called within the views, I now think that this is fine. `<%= link_to_user(foo) %>` is hard to improve upon, and I definitely agree with not asking view templates to do too much calculation.

With this, I think my concern has been suitably addressed - thanks for talking through this with me.

Reply:
Another approach that might be less confusing is if we remove the guard clause and change the language in the else condition:

```
def link_to_user(user_id)
  if (user = User.find_by(id: user_id))
    link_to user.kerberos_id, edit_admin_user_path(user.id)
  else
    "ID #{user_id} not an active user."
  end
end
```

That way, it will always return something, and it might be clearer to end users when the UI is trying to render a link for a system-generated record. What do you think?

Reply:
Ooh... I kind of like this. It makes the function more content-agnostic rather than checking for a bespoke value, and always returns _something_. If that is acceptable to you in how it appears to the user, then I think this could be a winner.